File: languages/cpp/app_templates/kapp/app_client.cpp
%{CPP_TEMPLATE}

#include <kapplication.h>
#include <dcopclient.h>
#include <qdatastream.h>
#include <qstring.h>

int main(int argc, char **argv)
{
    KApplication app(argc, argv, "%{APPNAMELC}_client", false);

    // get our DCOP client and attach so that we may use it
    DCOPClient *client = app.dcopClient();
    client->attach();

    // do a 'send' for now
    QByteArray data;
    QDataStream ds(data, IO_WriteOnly);
    if (argc > 1)
        ds << QString(argv[1]);
    else
        ds << QString("http://www.kde.org");
    client->send("%{APPNAMELC}", "%{APPNAME}Iface", "openURL(QString)", data);

    return app.exec();
}

File: languages/ruby/app_templates/kapp/app_client.rb
require 'Korundum'
include KDE

about = AboutData.new("%{APPNAMELC}_client", "%{APPNAME} client", "0.1")
CmdLineArgs.init(ARGV, about)

app = Application.new

# get our DCOP client and attach so that we may use it
client = app.dcopClient
client.attach

# do a 'send' for now
dcopRef = DCOPRef.new("%{APPNAMELC}", "%{APPNAME}Iface")
dcopRef.openURL("http://www.kde.org")

app.exec

